samples/app-ReporterPlus/ReporterPlus/Bots/ReporterPlusBot.cs

```diff
@@ -0,0 +1,217 @@
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using ReporterPlus.Models;
+using ReporterPlus.Helpers;
+
+namespace ReporterPlus.Bots
+{
+    public class ReporterPlusBot : TeamsActivityHandler
+    {
+        public ReporterPlusBot(IConfiguration configuration) : base()
+        {
+            Constants.MicrosoftAppId = configuration["MicrosoftAppId"];
+            Constants.MicrosoftAppPassword = configuration["MicrosoftAppPassword"];
+            Constants.TenantId = configuration["TenantId"];
+            Constants.BaseUrl = configuration["BaseUrl"];
+            Constants.ServiceUrl = configuration["ServiceUrl"];
+            Constants.OriginatorId = configuration["OriginatorId"];
+            Constants.SenderEmail = configuration["SenderEmail"];
+            Constants.BlobContainerName = configuration["BlobContainerName"];
+            Constants.BlobConnectionString = configuration["BlobConnectionString"];
+        }
+
+        protected override async Task<InvokeResponse> OnInvokeActivityAsync(ITurnContext<IInvokeActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.Name == "composeExtension/fetchTask")
+            {
+                MessagingExtensionAction action = new MessagingExtensionAction();
+                var actionJson = JsonConvert.DeserializeObject<MessagingExtensionActionDeserializer>(turnContext.Activity.Value.ToString());
+                action.CommandId = actionJson.commandId;
+                var task = await this.OnTeamsMessagingExtensionFetchTaskAsync(turnContext, action, cancellationToken);
+                return CreateInvokeResponse(task);
+            }
+            else if (turnContext.Activity.Name == "composeExtension/submitAction")
+            {
+                MessagingExtensionAction action = new MessagingExtensionAction();
+                var task = await this.OnTeamsMessagingExtensionSubmitActionAsync(turnContext, action, cancellationToken);
+                return CreateInvokeResponse(task);
+            }
+            else if (turnContext.Activity.Name == "task/fetch")
+            {
+                var actionJson = JsonConvert.DeserializeObject<ActionBase>(turnContext.Activity.Value.ToString());
+                var task = await ReturnViewDetails(actionJson.data.reqId);
+                return CreateInvokeResponse(task);
+            }
+            else if (turnContext.Activity.Name == "adaptiveCard/action")
+            {
+                var data = JsonConvert.DeserializeObject<ActionType>(turnContext.Activity.Value.ToString());
+                string channel = turnContext.Activity.ChannelId;
+                string verb = data.action.verb;
+                var blobInfo = BlobHelper.GetBlob(data.action.data.reqId, verb).Result;
+                string cardJsonString;
+
+                if (verb == "Refresh" && blobInfo.status == "Pending")
```

Comment:
I can see that there are 4 states/status that you have in the flow which are BaseCard, Pending, Approved, Rejected. I would suggest you to create a new class or enum for it. You will be able to update all the usages for this in ReporterPlusBot.cs, CardHandler.cs and other files where these are used.